src/runtime_src/core/common/detail/windows/sysinfo.h

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+// Local - Include files
+#include "core/common/sysinfo.h"
+#include "core/common/error.h"
+
+// System - Include Files
+#include <windows.h>
+#include <string>
+#include <thread>
+
+// 3rd Party Library - Include Files
+#include <boost/property_tree/ptree.hpp>
+#include <boost/format.hpp>
+
+namespace {
+
+static std::string
+getmachinename()
+{
+  std::string machine;
+  SYSTEM_INFO sysInfo;
+
+  // Get hardware info
+  ZeroMemory(&sysInfo, sizeof(SYSTEM_INFO));
+  GetSystemInfo(&sysInfo);
+  // Set processor architecture
+  switch (sysInfo.wProcessorArchitecture) {
+  case PROCESSOR_ARCHITECTURE_AMD64:
+    machine = "x86_64";
+    break;
+  case PROCESSOR_ARCHITECTURE_IA64:
+    machine = "ia64";
+    break;
+  case PROCESSOR_ARCHITECTURE_INTEL:
+    machine = "x86";
+    break;
+  case PROCESSOR_ARCHITECTURE_UNKNOWN:
+  default:
+    machine = "unknown";
+    break;
+  }
+
+  return machine;
+}
+
+static std::string
+osNameImpl()
+{
+  OSVERSIONINFO vi;
+  vi.dwOSVersionInfoSize = sizeof(vi);
+  if (GetVersionEx(&vi) == 0)
+    throw xrt_core::error("Cannot get OS version information");
+  switch (vi.dwPlatformId)
+  {
+  case VER_PLATFORM_WIN32s:
+      return "Windows 3.x";
+  case VER_PLATFORM_WIN32_WINDOWS:
+      return vi.dwMinorVersion == 0 ? "Windows 95" : "Windows 98";
+  case VER_PLATFORM_WIN32_NT:
+      return "Windows NT";
+  default:
+      return "Unknown";
+  }
+}
+
+static std::string
+get_bios_version() {
+  struct SMBIOSData {
+    uint8_t  Used20CallingMethod;
+    uint8_t  SMBIOSMajorVersion;
+    uint8_t  SMBIOSMinorVersion;
+    uint8_t  DmiRevision;
+    uint32_t  Length;
+    uint8_t  SMBIOSTableData[1];
+  };
+
+  SMBIOSData *bios_data = nullptr;
+  DWORD bios_size = GetSystemFirmwareTable('RSMB', 0, NULL, 0);
+  if (bios_size > 0) {
+    bios_data = (SMBIOSData*)malloc(bios_size);
+    // Retrieve the SMBIOS table
```

Comment:
malloc --> new

Reply:
I missed this one.  Let's `std::vector<char> bios_data{bios_size};`

Reply:
Thanks for pointing this out. Fixed